### PR TITLE
Try catch fix v2 and version update

### PR DESCRIPTION
**Pull request by zhenghao104**
**Task name**: <Name of changed or new pipeline task>

**Description**: Addressing Bug [1668409](https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/1668409/): Npm auth task should fail if location service call fails

Previous [fix ](https://github.com/microsoft/azure-pipelines-tasks/pull/9407) was [reverted ](https://github.com/microsoft/azure-pipelines-tasks/pull/9490) due to some proxy issues which had been fixed.

The new fix addresses the issue where the task should fail if it fails to retrieve packaging urls.

**Documentation changes required:** (Y/N) <Please mark if documentation changes are required>

**Added unit tests:** (Y/N) <Please mark if unit tests were added or updated according changes>

**Attached related issue:** (Y/N) <Please add link to related issue here>

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/DotNetCoreCLIV2/restorecommand.ts [hunk 1]
```diff
@@ -17,13 +17,9 @@ export async function run(): Promise<void> {
...
     } catch (error) {
-        tl.debug('Unable to get packaging URIs, using default collection URI');
+        tl.debug('Unable to get packaging URIs');
         logError(error);
-        const collectionUrl = tl.getVariable('System.TeamFoundationCollectionUri');
-        packagingLocation = {
-            PackagingUris: [collectionUrl],
-            DefaultPackagingUri: collectionUrl
-        };
+        throw new Error(error);
```
Comment (jmyersmsft) [inline Tasks/DotNetCoreCLIV2/restorecommand.ts:22]: I'm not as much of a TypeScript/JavaScript expert as others on the team, but wrapping the exception in a new Error object seems suspicious. I'd think you would want to simply rethrow the same object in these cases:

```suggestion
        throw error;
```

Same comment in the exception-rethrowing spots too

Comment (zhenghao104) [reply Tasks/DotNetCoreCLIV2/restorecommand.ts:22]: That makes sense, didnt think about that, @magleaso @satbai any thoughts on this?

Or we can define a string something like [this ](https://github.com/microsoft/azure-pipelines-tasks/blob/7bf2391069f629f64fc18f4258995565283ce37b/Tasks/PipAuthenticateV0/task.json#L83) in each task?

File: Tasks/NpmAuthenticateV0/task.loc.json [hunk 2]
```diff
@@ -66,4 +67,4 @@
       "argumentFormat": ""
     }
   }
-}
+}
```
Comment (magleaso) [inline Tasks/NpmAuthenticateV0/task.loc.json:70]: Nit: removed newline

File: Tasks/NpmV1/task.loc.json [hunk 3]
```diff
@@ -197,4 +197,4 @@
     "WorkingDirectoryNotDirectory": "ms-resource:loc.messages.WorkingDirectoryNotDirectory",
     "NGCommon_AreaNotFoundInSps": "ms-resource:loc.messages.NGCommon_AreaNotFoundInSps"
   }
-}
+}
```
Comment (magleaso) [inline Tasks/NpmV1/task.loc.json:200]: nit: newline

File: Tasks/NuGetV0/task.loc.json [hunk 4]
```diff
@@ -75,4 +75,4 @@
     "NGCommon_UnableToFindTool": "ms-resource:loc.messages.NGCommon_UnableToFindTool",
     "Warning_UpdatingNuGetVersion": "ms-resource:loc.messages.Warning_UpdatingNuGetVersion"
   }
-}
+}
```
Comment (magleaso) [inline Tasks/NuGetV0/task.loc.json:78]: nit: newline

Comment (zhenghao104) [reply Tasks/NuGetV0/task.loc.json:78]: This is generated when i build the task, what does nit mean?

File: Tasks/DotNetCoreCLIV2/restorecommand.ts (hunk 1 above)
Comment (magleaso) [reply Tasks/DotNetCoreCLIV2/restorecommand.ts:22]: Jonathan's comment makes sense to me. Not sure if that messages property is for localization. If not, I don't think it needs to be changed. If the message is only used in one place per task, it's just more readable to put the string in directly.

File: Tasks/NuGetV0/task.loc.json (hunk 4 above)
Comment (magleaso) [reply Tasks/NuGetV0/task.loc.json:78]: It's short for nitpick. It means minor recommendation

File: Tasks/NpmAuthenticateV0/task.loc.json [hunk 5]
```diff
@@ -23,7 +23,8 @@
       "name": "workingFile",
       "label": "ms-resource:loc.input.label.workingFile",
       "helpMarkDown": "ms-resource:loc.input.help.workingFile",
-      "type": "filePath"
+      "type": "filePath",
+      "required": true
```
Comment (magleaso) [inline Tasks/NpmAuthenticateV0/task.loc.json:27]: Is this change intended to be in this PR? It seems good based on the docs, but I haven't checked the task code to see if it handles the case w/ no workingFile. This could be a breaking change if so

File: Tasks/NuGetV0/task.loc.json (hunk 4 above)
Comment (zhenghao104) [reply Tasks/NuGetV0/task.loc.json:78]: I see, yeah i didnt change anything in the original task.json and this change is auto generated, not sure if it's because one of the older commits did not check the correct changes in.

File: Tasks/NpmAuthenticateV0/task.loc.json (hunk 5 above)
Comment (zhenghao104) [reply Tasks/NpmAuthenticateV0/task.loc.json:27]: Right, it's not intended to be this PR, it's auto generated, if it shouldn't be here i will have to remove it manually after task.loc.json is generated (together with all those new lines problem?)

Comment (magleaso) [reply Tasks/NpmAuthenticateV0/task.loc.json:27]: If they're auto-generated, just keep them imo. This input should be required per the docs

File: Tasks/DotNetCoreCLIV2/restorecommand.ts (hunk 1 above)
Comment (zhenghao104) [reply Tasks/DotNetCoreCLIV2/restorecommand.ts:22]: Since we are already logging the error, I will just change it to `throw error` like Jonathan suggested

File: Tasks/NpmAuthenticateV0/task.loc.json (hunk 2 above)
Comment (zhenghao104) [reply Tasks/NpmAuthenticateV0/task.loc.json:69]: Leaving the auto generated new line change in the PR.

File: Tasks/DotNetCoreCLIV2/restorecommand.ts (hunk 1 above)
Comment (zhenghao104) [reply Tasks/DotNetCoreCLIV2/restorecommand.ts]: Fixed

File: Tasks/NpmAuthenticateV0/task.loc.json (hunk 5 above)
Comment (zhenghao104) [reply Tasks/NpmAuthenticateV0/task.loc.json]: Reverted the change to keep the PR clean of potential breaking changes

**Review by jmyersmsft — APPROVED** [review]
